### DISCUSSION TITLE
useCalendar - navigation button props contain an (unknown?) onFocusChange prop

### DISCUSSION THREAD

**Discussion by PrettyCoffee** [adobe/react-spectrum#5170] [Q&A] [answered]
In the latest release of "@react-aria/calendar", the returned button props of `useCalendar` include `onFocusChange` instead of `onBlur` and `onFocus`.
[According commit](https://github.com/adobe/react-spectrum/commit/06e63d3ae7898b8cd81fdb6fa87b763faf2054bf#diff-39e5de9f671c37a9382b995f347a49883fa876e6e19933df082dedfcdefc66b6)

This produces console errors in reacts dev mode:
> Warning: Unknown event handler property `onFocusChange`. It will be ignored.

This happens for me when spreading `prevButtonProps` or `nextButtonProps` into a button component, like proposed in the [documentation example](https://react-spectrum.adobe.com/react-aria/useCalendar.html#example).

Furthermore, I could not find any documentation on a `onFocusChange` react prop, so I was wondering if this was done by mistake? Or am I doing something wrong?

**Comment by PrettyCoffee** [comment]
Small repro where only the navigation buttons are rendered:
https://codesandbox.io/p/sandbox/practical-sara-fyxtmc?file=%2Fsrc%2FApp.tsx%3A3%2C19

-> See the browser devtools console
![image](https://github.com/adobe/react-spectrum/assets/66521452/027e8280-b32b-43a6-b739-ca0f58c507c1)

**Comment by reidbarber** [comment, marked as answer]
You'll need to spread `prevButtonProps`/ `nextButtonProps` onto a button created with [`useButton`](https://react-spectrum.adobe.com/react-aria/useButton.html). You can update your example to this instead:

```jsx
function Button(props) {
  const ref = useRef(null);
  const { buttonProps } = useButton(props, ref);
  return <button {...buttonProps} ref={ref}>{props.children}</button>;
}
```

https://codesandbox.io/p/sandbox/hungry-mccarthy-pjdm4y

This is because `prevButtonProps`/ `nextButtonProps` contain handlers such as `onPress` and `onFocusChange` that aren't attributes of native buttons.

**Reply by PrettyCoffee** [reply]
Aah, I see, that makes much more sense now! Thanks a lot for the quick answer :)
